test/reduce_test.cpp

```diff
@@ -850,12 +850,11 @@ struct reduce_driver : test_driver
         };
 
         // Special data generation for NORM1 and NORM2 using a space of limitless number of values.
-        // This method is slower due to the use of GET_RAND(), it is usually used for manual
-        // testing
         auto gen_value_3 = [&](auto... is) {
-            auto rand_upper   = tensor_elem_gen_integer{max_value}(is...);
-            auto sign_value   = tensor_elem_gen_checkboard_sign{}(is...);
-            double rand_ratio = static_cast<double>(GET_RAND() / (static_cast<double>(RAND_MAX)));
+            auto rand_upper = tensor_elem_gen_integer{max_value}(is...);
+            auto sign_value = tensor_elem_gen_checkboard_sign{}(is...);
+            auto rand_ratio = prng::gen_A_to_B(
+                0.1, 1.); // limit range due to numeric errors, see WORKAROUND_GPU_NUMERIC_ERROR
```

Comment:
Where can I see this W/A?

Reply:
It's a define in this file, which disables the test completely.
The problem is still existing and accidentally it was just hidden for some time.

Reply:
@CAHEK7 Sorry but I am unable to see how value the of rand_ratio is related to WORKAROUND_GPU_NUMERIC_ERROR.

Also it seems to me that the range of rand_ratio was not [0.1 ... 1.0] before this change. Can you please check or justify the change?

Reply:
@atamazov 
This test was disabled completely by `WORKAROUND_GPU_NUMERIC_ERROR` due to numerical instability. For some reasons the problem has gone but it was not fixed since it is so sensitive to the input data. From the other hand loss of precision is a quite common problem for reduction of big amount of data, when tiny values started to be ignored.
So there are:
1) reduction without loss of precision mitigation - it can be ok or not, depends on the API precision specification or guarantees
2) the test generates the data which is out of scope of the algorithm - it's not ok, because even algorithms with loss of precision mitigation can be broken by specially prepared data
3) the test uses higher precision algorithm for verification - it's not ok, because loss of precision can't be mitigated by epsilon comparison

I've chosen 2 and excluded small values from the input data.
There is also 2.5 option when limited range can be applied to that specific test like it was done by `WORKAROUND_GPU_NUMERIC_ERROR` but do not disable the test.

We can either get an API precision specification and fix reduction algorithm, and\or we can use the same precision for verification, but this PR is not got to do any of this.

Reply:
@CAHEK7 
> This test was disabled completely by WORKAROUND_GPU_NUMERIC_ERROR due to numerical instability

To clarify: we do not know the reason of the issue (and I do not remember how it manifested itself... likely a validation failure, but I am not sure about instability). I tend to blame the compiler for GPU.

> the test uses higher precision algorithm for verification - it's not ok, because loss of precision can't be mitigated by epsilon comparison

Let's discuss this 1:1.

> the test generates the data which is out of scope of the algorithm - it's not ok, because even algorithms with loss of precision mitigation can be broken by specially prepared data

This is important observation. So the main reason of this change is putting the input data back into the valid scope of the algorithm, right? Very good finding.

I recommend removing the comment about WORKAROUND_GPU_NUMERIC_ERROR from the source code, as we can neither prove nor disprove that it relates to the input range. Thanks.